### DISCUSSION TITLE
Support for `new_version` in TOML configuration

### DISCUSSION THREAD

**Discussion by HenriBlacksmith** [callowayproject/bump-my-version#149] [General]
My team just moved from `bump2version` to `bump-my-version` and we noted that it is not possible to force the `new_version` value when using TOML configuration.

Our workflow was to have a single GitLab CI job to bump the version on our main branch and if one MR since last tag/version was adding the `new_version` in the `.bumpversion.cfg` file (to force a minor or major bump because of the nature of the changes), then the job was doing the bump.

In `bump-my-version` it seems that this option is just on the CLI? Was this intentional and what is the intent (I am not able to find traces of this change, but I did  not do a deep search either ...)

**Comment by coordt** [comment]
This was an intended change since having both `current_version` and `new_version` in a configuration was deemed confusing.

Another option might be to use a `.env` file, like `new_version.env` with the contents `export BUMPVERSION_NEW_VERSION=<newversion>`

Then you could source that file to set up the environment variable when you run the the bumpversion command.

I'm happy to discuss other workarounds or workflows with you.

**Reply by HenriBlacksmith** [reply]
Looks like an alternative yes, for now we moved the complexity to the CI (using one job for each bump type)

Injecting an environment variable in the job is also possible (either via GitLab job inputs or as you proposed)
